wasmplugin/plugin.go

```diff
@@ -236,29 +236,33 @@ func (ctx *httpContext) OnHttpRequestHeaders(numHeaders int, endOfStream bool) t
 	ctx.metrics.CountTX()
 
 	authority, err := proxywasm.GetHttpRequestHeader(":authority")
-	if err == nil {
-		if waf, isDefault, resolveWAFErr := ctx.perAuthorityWAFs.getWAFOrDefault(authority); resolveWAFErr == nil {
-			ctx.tx = waf.NewTransaction()
+	if err != nil {
+		proxywasm.LogWarnf("Failed to get the :authority pseudo-header: %v", err)
```

Comment:
This can be debug since in servers that populate the property and not the header, it's expected behavior. If both aren't populated, it is an error which I think we catch below

Reply:
updated again. Please review